### PR TITLE
Clarify retry policies

### PR DESCRIPTION
**Pull request by tysonmote**
### Proposed changes

This commit makes the verbiage around our retry policies more accurate because claims around attempt counts are currently out-of-date.

### Merge timing

When approved.

### Related issues (optional)

N/A

### CODE REVIEW

File: src/connections/functions/source-functions.md [hunk 1]
```diff
@@ -385,7 +385,7 @@ Copy and paste this URL into the upstream tool or service to send data to this s
 
 ##### What is the retry policy for a webhook payload?
 
-The webhook payload retries up to six times with an exponential backoff for the function in the event of a failure with the function. After six attempts, the message is dropped.
+Segment retries invocations that throw RetryError or Timeout errors up to 6 times. After 6 attempts, the request is dropped.
```
Comment (markzegarelli) [inline src/connections/functions/source-functions.md:388]:
```suggestion
Segment retries invocations that throw RetryError or Timeout errors up to 6 times. After six attempts, the request is dropped.
```

File: src/partners/faqs.md [hunk 2]
```diff
@@ -68,6 +68,4 @@ You can subscribe to updates [here](https://status.segment.com/).
 
 ### Does Segment retry data?
 
-Segment retries nine times over the course of four hours. This will increase the number of attempts for messages, so we'll try and re-deliver them another 4 times after some backoff.
-
-We don't retry anywhere which is the sign of an expired API key or failed payment. However, if we push bad code that results in a malformed payload and a 400 or 422 response from an endpoint, we also won't retry given that the call would not ever succeed.
+Segment retries requests for up to 4 hours if they fail due to temporary errors like timeouts, 429 Too Many Request status codes, and 5xx status codes. Segment does not retry requests that fail due to permanent errors like authentication failures or 4xx status codes.
```
Comment (markzegarelli) [inline src/partners/faqs.md:71]:
```suggestion
Segment retries requests for up to four hours if they fail due to temporary errors like timeouts, 429 Too Many Request status codes, and 5xx status codes. Segment does not retry requests that fail due to permanent errors like authentication failures or 4xx status codes.
```

**Review by markzegarelli — CHANGES_REQUESTED** [review]
Thanks @tysonmote!

**Review by markzegarelli — APPROVED** [review]

File: src/connections/destinations/catalog/webhooks/index.md [hunk 3]
```diff
@@ -226,4 +226,4 @@ Under 'Connection Settings', you can provide up to 5 webhooks.
 
 ### Retries
 
-Our webhooks destination will retry any request that returns 5xx errors, multiple times, for a maximum of 4 hours.
+Segment will retry any request that fails due to temporary errors like timeouts and 5xx status codes for up to four hours.
```
Comment (stayseesong) [inline src/connections/destinations/catalog/webhooks/index.md:229]:
```suggestion
Segment retries any request that fails due to temporary errors like timeouts and 5xx status codes for up to four hours.
```

**Review by stayseesong — APPROVED** [review]

**Pull request merged**